SplunkRumWorkspace/SplunkRum/SplunkRum/ConnectionInstrumentation.swift

```diff
@@ -0,0 +1,190 @@
+                    let valStr = val as? String
+                    if valStr != nil {
+                        if valStr!.starts(with: "traceparent") {
+                            addLinkToSpan(span: span, valStr: valStr!)
+                        }
+                    }
+                }
+            }
+        }
+    }
+    if error != nil {
+        span.setAttribute(key: "error", value: true)
+        span.setAttribute(key: "exception.message", value: error!.localizedDescription)
+        span.setAttribute(key: "exception.type", value: String(describing: type(of: error!)))
+    }
+    span.setAttribute(key: "http.response_content_length_uncompressed",
+                      value: Int(hresponse?.expectedContentLength ?? 0))
+    if hostConnectionType != nil {
+        span.setAttribute(key: "net.host.connection.type", value: hostConnectionType!)
+    }
+    span.end()
+}
+func swizzleClassMethod(clazz: AnyClass, orig: Selector, swizzled: Selector) {
+    let origM = class_getClassMethod(clazz, orig)
+    let swizM = class_getClassMethod(clazz, swizzled)
+    if origM != nil && swizM != nil {
+        method_exchangeImplementations(origM!, swizM!)
+    } else {
+        debug_log("warning: could not swizzle "+NSStringFromSelector(orig))
+    }
+}
+func initalizeConnectionInstrumentation() {
+    let connection = NSURLConnection.self
+    swizzleClassMethod(clazz: connection,
+                       orig: #selector(NSURLConnection
+                                        .sendAsynchronousRequest(_:queue:completionHandler:)),
+                       swizzled: #selector((NSURLConnection.splunk_swizzled_connection_sendAsynchronousRequest(request:queue:completionHandler:))))
+    swizzleClassMethod(clazz: connection,
+                       orig: #selector(NSURLConnection
+                                        .sendSynchronousRequest(_:returning:)),
+                       swizzled: #selector(NSURLConnection.splunk_swizzled_connection_sendSynchronousRequest(_:returning:)))
+}
+
+// Convert from NSData to json object
+public func dataToJSON(data: Data) -> Any? {
+     guard let deserializedValues = try? JSONSerialization
+            .jsonObject(with: data,
+                        options: JSONSerialization
+                            .ReadingOptions.mutableContainers) else { return false}
+        return deserializedValues
+}
+
+// swiftlint:disable missing_docs
+extension NSURLConnection {
+    @objc open class func
+    splunk_swizzled_connection_sendSynchronousRequest
+    (_ request: URLRequest?, returning response: AutoreleasingUnsafeMutablePointer<URLResponse?> )
+    throws -> Data {
+        print("inside swizzle sendSynchronous method")
+        let span = startConnectionSpan(request: request)
+        var status = "Sucess"
+        var data = Data()
+        do {
+            data = try splunk_swizzled_connection_sendSynchronousRequest(request, returning: response)
```

Comment:
Since this is an Obj-C function called from Swift, is it necessary for the try catch here?

Reply:
yes, it is necessary to use try catch because method throws data , so to handle it we must use it.